onnxruntime/contrib_ops/cpu/transformers/beam_search.cc

```diff
@@ -108,9 +110,11 @@ Status BeamSearch::SetupSubgraphExecutionInfo(const SessionState& session_state,
     } else if (attribute_name == "decoder") {
       ORT_ENFORCE(t5_decoder_subgraph_ == nullptr,
                   "SetupSubgraphExecutionInfo should only be called once for each subgraph.");
+      bool has_hidden_states = parameters_.model_type == IBeamSearchParameters::kModelTypeZcode ? false : true;
```

Comment:
We shall be able to detect this based on subgraph inputs. In this way, it could be more general (no binding to a model type).